File: codegen/results/v002/m3-max-48gb/ollama/qwen3.8_27b-mlx_nothink/20260910_072053/typescript/005_unit_test_writer/solution.ts
function calculateDiscount(price: number, quantity: number): number {
    if (quantity < 10) return price * quantity;
    else if (quantity < 50) return price * quantity * 0.9;
    else return price * quantity * 0.75;
}

interface TestCase {
    description: string;
    price: number;
    quantity: number;
    expected: number;
}

const testCases: TestCase[] = [
    {
        description: "Quantity 9 (no discount)",
        price: 10,
        quantity: 9,
        expected: 10 * 9, // 90
    },
    {
        description: "Quantity 10 (10% discount)",
        price: 10,
        quantity: 10,
        expected: 10 * 10 * 0.9, // 90
    },
    {
        description: "Quantity 49 (10% discount)",
        price: 10,
        quantity: 49,
        expected: 10 * 49 * 0.9, // 441
    },
    {
        description: "Quantity 50 (25% discount)",
        price: 10,
        quantity: 50,
        expected: 10 * 50 * 0.75, // 375
    },
    {
        description: "Quantity 1 (no discount, single item)",
        price: 5,
        quantity: 1,
        expected: 5 * 1, // 5
    },
    {
        description: "Quantity 100 (25% discount, large quantity)",
        price: 20,
        quantity: 100,
        expected: 20 * 100 * 0.75, // 1500
    },
];

for (const test of testCases) {
    const result = calculateDiscount(test.price, test.quantity);
    // Use a small epsilon for floating point comparison
    if (Math.abs(result - test.expected) < 1e-9) {
        console.log(`PASS: ${test.description}`);
    } else {
        console.log(`FAIL: ${test.description} (expected: ${test.expected}, got: ${result})`);
    }
}

File: codegen/results/v002/m3-max-48gb/ollama/qwen3.8_27b-mlx_nothink/20260910_072053/go/005_unit_test_writer/solution.go
package main

import (
	"fmt"
	"math"
)

func calculateDiscount(price float64, quantity int) float64 {
	if quantity < 10 {
		return price * float64(quantity)
	} else if quantity < 50 {
		return price * float64(quantity) * 0.9
	}
	return price * float64(quantity) * 0.75
}

func approxEqual(a, b float64) bool {
	return math.Abs(a-b) < 1e-9
}

func main() {
	// Test case 1: quantity = 9 (below first threshold, no discount)
	// Expected: 10.0 * 9 = 90.0
	expected1 := 90.0
	got1 := calculateDiscount(10.0, 9)
	if approxEqual(got1, expected1) {
		fmt.Println("PASS: quantity 9 (no discount)")
	} else {
		fmt.Printf("FAIL: quantity 9 (no discount) (expected: %f, got: %f)\n", expected1, got1)
	}

	// Test case 2: quantity = 10 (at boundary, 10% discount)
	// Expected: 10.0 * 10 * 0.9 = 90.0
	expected2 := 90.0
	got2 := calculateDiscount(10.0, 10)
	if approxEqual(got2, expected2) {
		fmt.Println("PASS: quantity 10 (10%% discount boundary)")
	} else {
		fmt.Printf("FAIL: quantity 10 (10%% discount boundary) (expected: %f, got: %f)\n", expected2, got2)
	}

	// Test case 3: quantity = 49 (below 50 threshold, 10% discount)
	// Expected: 20.0 * 49 * 0.9 = 882.0
	expected3 := 882.0
	got3 := calculateDiscount(20.0, 49)
	if approxEqual(got3, expected3) {
		fmt.Println("PASS: quantity 49 (10%% discount)")
	} else {
		fmt.Printf("FAIL: quantity 49 (10%% discount) (expected: %f, got: %f)\n", expected3, got3)
	}

	// Test case 4: quantity = 50 (at boundary, 25% discount)
	// Expected: 20.0 * 50 * 0.75 = 750.0
	expected4 := 750.0
	got4 := calculateDiscount(20.0, 50)
	if approxEqual(got4, expected4) {
		fmt.Println("PASS: quantity 50 (25%% discount boundary)")
	} else {
		fmt.Printf("FAIL: quantity 50 (25%% discount boundary) (expected: %f, got: %f)\n", expected4, got4)
	}
}